core/test/unit/models/post_spec.js

```diff
@@ -1890,4 +1890,84 @@ describe('Unit: models/post', function () {
             });
         });
     });
+
+    describe('Mobiledoc conversion', function () {
+        let config = require('../../../server/config');
+        let origConfig = _.cloneDeep(config);
+        let labs = require('../../../server/services/labs');
+        let origLabs = _.cloneDeep(labs);
+        let sandbox = sinon.sandbox.create();
+        let events;
+
+        beforeEach(function () {
+            events = {
+                post: []
+            };
+
+            sandbox.stub(models.Post.prototype, 'emitChange').callsFake(function (event) {
+                events.post.push({event: event, data: this.toJSON()});
+            });
+        });
+
+        afterEach(function () {
+            sandbox.restore();
+        });
+
+        it('uses v2 if Koenig is enabled', function () {
+            sandbox.stub(config, 'get').callsFake(function (key) {
+                if (key === 'enableDeveloperExperiments') {
+                    return true;
+                }
+                return origConfig.get(key);
+            });
+            sandbox.stub(labs, 'isSet').callsFake(function (key) {
+                if (key === 'koenigEditor') {
+                    return true;
+                }
+                return origLabs.get(key);
+            });
+
+            let newPost = testUtils.DataGenerator.forModel.posts[2];
+
+            return models.Post.add(
+                newPost,
+                _.merge(
+                    {withRelated: ['author']},
+                    testUtils.context.editor
+                )
+            ).then((post) => {
+                should.exist(post);
+                post.has('html').should.equal(true);
+                post.get('html').should.equal('<div class="kg-post">\n<h2 id="testing">testing</h2>\n<p>mctesters</p>\n<ul>\n<li>test</li>\n<li>line</li>\n<li>items</li>\n</ul>\n\n</div>');
+            });
+        });
+
+        it('uses v2 if Koenig is disabled but post is not v1 compatible', function () {
+            let newPost = testUtils.DataGenerator.forModel.posts[2];
+
+            newPost.mobiledoc = JSON.stringify({
+                version: '0.3.1',
+                atoms: [],
+                cards: [],
+                markups: [],
+                sections: [
+                    [1, 'p', [
+                        [0, [], 0, 'Test']
+                    ]]
+                ]
+            });
+
+            return models.Post.add(
+                newPost,
+                _.merge(
+                    {withRelated: ['author']},
```

Comment:
Why `withRelated: ['author']`?

Reply:
It was in other tests which weren't testing the author so I assumed it was necessary. I'll remove.